tests/end2end/features/downloads.feature

```diff
@@ -22,6 +22,27 @@ Feature: Downloading things from a website.
         And I wait until the download is finished
         Then the downloaded file download.bin should exist
 
+    Scenario: Using :download with no URL
+        When I set storage -> prompt-download-directory to false
+        And I open data/downloads/downloads.html
+        And I run :download
+        And I wait until the download is finished
+        Then the downloaded file Simple downloads.html should exist
+
+    Scenario: Using :download with no URL on an image
+        When I set storage -> prompt-download-directory to false
+        And I open data/downloads/qutebrowser.png
+        And I run :download
+        And I wait until the download is finished
+        Then the downloaded file qutebrowser.png should exist
+
+    Scenario: Using :download with no URL and no page title
```

Comment:
It looks like QtWebEngine autogenerates a filename while QtWebKit doesn't, which also explains that you don't see it locally. After adding some unittests (see my review comment), I'd suggest just removing this end2end test entirely (but keep the other two).